docs/markdown/Using Pants/environments.md

```diff
@@ -0,0 +1,221 @@
+>
+> As we gain more experience with how environments are used in the wild, it's possible that more static validation can be added: your feedback would be very welcome!
+
+### Setting the environment on many targets at once
+
+To use an environment everywhere in your repository (or only within a particular subdirectory, or with a particular target
+type), you can use the [`__defaults__` builtin](doc:targets#field-default-values). For example, to use an environment named `my_default_environment` globally by default, you would add the following to a `BUILD` file at the root of the repository:
+```python
+__defaults__(all=dict(environment="my_default_environment"))
+```
+... and individual targets could override the default as needed.
+
+### Environment matching
+
+A single environment name may end up referring to different environment targets on different physical machines, or with different global settings applied: this is known as environment "matching".
+
+* `local_environment` targets will match if their `compatible_platforms=` field matches localhost's platform.
+* `docker_environment` targets will match [if Docker is enabled](doc:reference-global#docker_execution), and if their `platform=` field is compatible with localhost's platform.
+* `remote_environment` targets will match [if Remote execution is enabled](doc:reference-global#remote_execution).
+
+It a particular environment target _doesn't_ match, it can configure a `fallback_environment=` which will be attempted next. This allows for forming preference chains which are referred to by whichever environment name is at the head of the chain.
+
+For example: a chain like "prefer remote execution if enabled, but fall back to local execution if the platform matches, otherwise use docker" might be configured via the targets:
+```python
+remote_environment(
+  name="remote",
+  fallback_environment="local",
+  ..
+)
+
+local_environment(
+  name="local",
+  compatible_platforms=["linux_x86_64"],
+  fallback_environment="docker",
+)
+
+docker_environment(
+  name="docker",
+  ..
+)
+```
+
+In future versions, environment targets will gain additional predicates to control whether they match (for example: `local_environment` will likely gain a predicate that looks for the [presence or value of an environment variable](https://github.com/pantsbuild/pants/issues/17107). But in the meantime, it's possible to override which environments are matched for particular use cases by overriding their configured names: see the "Toggle use of an environment" workflow below for an example.
+
+## Example workflows
+
+### Enabling remote execution globally
+
+`remote_environment` targets match unless the [`--remote-execution`](doc:reference-global#remote_execution) option is disabled. So to cause a particular environment name to use remote execution whenever it is enabled, you could define environment targets which tried remote execution first, and then fell back to local execution:
```

Comment:
This is about half accurate: making it accurate by introducing a deprecation of the current behavior to resolve https://github.com/pantsbuild/pants/issues/11148 will be my first goal tomorrow.

Reply:
Ok, nevermind: it is currently accurate, and so #11148 is resolved! I will look into whether we need a deprecation, or at the very least a backwards-compatibility note somewhere in this doc.